docarray/array/mixins/match.py

```diff
@@ -70,20 +71,34 @@ def match(
         for d in self:
             d.matches.clear()
 
-        match_docs = darray.find(
-            self,
-            metric=metric,
-            limit=limit,
-            normalization=normalization,
-            metric_name=metric_name,
-            batch_size=batch_size,
-            exclude_self=exclude_self,
-            filter=filter,
-            only_id=only_id,
-            use_scipy=use_scipy,
-            device=device,
-            num_worker=num_worker,
-        )
+        if on is not None:
+            find_kwargs = {'on': on}
+        else:
+            find_kwargs = {}
+
+        try:
+            match_docs = darray.find(
+                self,
+                metric=metric,
+                limit=limit,
+                normalization=normalization,
+                metric_name=metric_name,
+                batch_size=batch_size,
+                exclude_self=exclude_self,
+                filter=filter,
+                only_id=only_id,
+                use_scipy=use_scipy,
+                device=device,
+                num_worker=num_worker,
+                **find_kwargs,
+            )
+        except TypeError as e:
+            if 'on' in str(e):
```

Comment:
I do not like this logic, the ValueError should already be sent from inside.

Also `on` is too common of a word

Reply:
I agree that this is not great. But if ValueError should be sent from the inside we would have to implement that for every backend. Alternatively we can just remove this check and it will also fail if, but in a less informative way.